src/cardanoWallet.ts

```diff
@@ -144,7 +144,7 @@ export async function startCardanoWallet(
             config.tlsConfiguration.svKey,
           ]
         : [],
-      config.smashUrl ? ['--smash-url', config.smashUrl] : [],
+      config.smashUrl ? ['--pool-metadata-fetching', config.smashUrl] : ['--pool-metadata-fetching', 'direct'],
```

Comment:
Please keep this in sync with cardano-wallet. So change the smashUrl config option to be a union type which models the Haskell type. Something like:

```typescript
export type PoolMetadataSource = "none" | "direct" | { smashUrl: string };
```

and then (with formatting tweaked obviously)

```suggestion
      config.poolMetadataSource
        ? [
            '--pool-metadata-fetching',
            typeof config.poolMetadataSource === 'string'
              ? config.poolMetadataSource
              : config.poolMetadataSource.smashUrl,
          ]
        : [],
```

Reply:
But by default, we want `direct` fetching, no?